include/coap/pdu.h

```diff
@@ -308,6 +417,15 @@ void coap_pdu_clear(coap_pdu_t *pdu, size_t size);
  */
 coap_pdu_t *coap_new_pdu(void);
 
+/**
+ * Creates a new CoAP PDU. The object is created on the heap and must be released
+ * using coap_delete_pdu();
+ *
+ * @deprecated This function allocates the maximum storage for each
```

Comment:
As the original function has been clearly marked as deprecated, I do not think that a new function should be introduced that mimics the deprecated behavior.